hometask-3

cvbnnthc1 opened the pull request.

cvbnnthc1 committed:
1-homework

cvbnnthc1 committed:
Update .gitignore

cvbnnthc1 committed:
Исправления по 1-й дз

cvbnnthc1 committed:
hometask-2

cvbnnthc1 committed:
hometask-3

cvbnnthc1 committed:
dz 1 исправление

cvbnnthc1 committed:
dz2 исправление, пришлосб хранить имя комнаты в двери, по-другому не придумал, как в actionable api можно отледить нужную дверь

cvbnnthc1 committed:
удалил лишний класс

cvbnnthc1 committed:
Merge branch 'hometask-1' into hometask-2

cvbnnthc1 committed:
Merge branch 'hometask-2' into hometask-3

cvbnnthc1 committed:
правка конфликтов dz-3

cvbnnthc1 committed:
Merge branch 'hometask-3' of https://github.com/cvbnnthc1/smart-home-2020 into hometask-3

cvbnnthc1 committed:
правки dz 3

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/Signalization.java:
```diff
@@ -0,0 +1,17 @@
+        return state;
+    }
+
+    public void setState(State state) {
```
package private

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/State.java:
```diff
@@ -0,0 +1,5 @@
+package ru.sbt.mipt.oop;
+
+public interface State {
```
SignalizationState

a-ivanov replied:
По заданию, тут должно быть 3 метода, доделайте.

cvbnnthc1 committed:
dz 2: door и light теперь Actionable

cvbnnthc1 committed:
Merge branch 'hometask-2' into hometask-3

cvbnnthc1 committed:
правки dz-3

Пришлось сделать SignalizationState абстрактным классом, чтобы везде не дублировать

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/Signalization.java:
```diff
@@ -0,0 +1,58 @@
+            if (((Activated) signalizationState).checkCode(event.getCode())) {
+                signalizationState.deactivate();
+            } else {
+                System.out.println("Wrong deactivation code");
```
Нет, здесь нарушение SRP. Логика проверки кода - ответственность состояний сигнализации.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/Application.java:
```diff
@@ -1,80 +1,31 @@
+        processors.add(new SignalizationEventProcessor(signalization));
+        ProcessingScript processor = new StandardProcessingScript(processors);
+        SensorEvent event = sensor.getNextSensorEvent();
         while (event != null) {
```
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%BF%D0%B5%D1%80%D0%B5%D0%BD%D0%B5%D1%81%D0%B8%D1%82%D0%B5-%D0%BB%D0%BE%D0%B3%D0%B8%D0%BA%D1%83-%D1%86%D0%B8%D0%BA%D0%BB%D0%B0-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B8-%D1%81%D0%BE%D0%B1%D1%8B%D1%82%D0%B8%D0%B9-%D0%B2-%D0%BE%D1%82%D0%B4%D0%B5%D0%BB%D1%8C%D0%BD%D1%8B%D0%B9-%D0%BA%D0%BB%D0%B0%D1%81%D1%81

cvbnnthc1 committed:
dz-3 исправления

cvbnnthc1 committed:
dz-3

cvbnnthc1 committed:
dz-3

cvbnnthc1 committed:
dz 3

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/Signalization.java:
```diff
@@ -0,0 +1,19 @@
+        }
+    }
+
+    public SignalizationState getState() {
```
Проверка состояния - это деталь реализации, к ней лучше не давать доступ в клиентском коде.
Захотите сменить реализацию с состояния на что-то другое - придется много кода менять.
Лучше в сигнализацию добавить методы для клиента типа `isAlarmed()`, внутри них уже проверять состояние.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/EventHandlerDecorator.java:
```diff
@@ -0,0 +1,23 @@
+package ru.sbt.mipt.oop;
+
+public class EventHandlerDecorator implements EventHandler {
```
Лучше SignalizationDecorator, раз тут логика завязана на сигнализацию.
